### PR TITLE
Add factory pattern

### PR DESCRIPTION
**Pull request by hanstirtaputra**
## Changes made:
1. Refactor `CanvasElementMapView` to apply factory pattern in view creation
2. Remove direct access of model from `CanvasElementMapView` by using a getter through `CanvasViewModel`

### CODE REVIEW

File: Dynavity/Dynavity/view/CanvasElementMapView.swift [hunk 1]
```diff
@@ -18,22 +12,7 @@ struct CanvasElementMapView: View {
     }
 
     private func transformToView(element: CanvasElementProtocol) -> some View {
```
Comment (sebastiantoh) [inline Dynavity/Dynavity/view/CanvasElementMapView.swift]: Do you think we should remove this function and directly use the view factory in the foreach?

**Review by sebastiantoh — DISMISSED** [review]
LGTM!

File: Dynavity/Dynavity/view/CanvasElementMapView.swift (hunk 1 above)
Comment (hanstirtaputra) [reply Dynavity/Dynavity/view/CanvasElementMapView.swift]: Sounds good! Thanks

**Review by sebastiantoh — APPROVED** [review]
lgtm

**Pull request merged**